outetts.js/version/v1/prompt_processor.js

```diff
@@ -0,0 +1,107 @@
+export class PromptProcessor {
+    /**
+     *
+     * @param {import('@huggingface/transformers').PreTrainedTokenizer} tokenizer
+     * @param {string[]} languages
+     */
+    constructor(tokenizer, languages) {
+        this.tokenizer = tokenizer;
+        this.bos = "<|im_start|>";
+        this.eos = "<|im_end|>";
+        this.special_tokens = {
+            audio_code: "<|{}|>",
+            text_start: "<|text_start|>",
+            text_end: "<|text_end|>",
+            audio_start: "<|audio_start|>",
+            audio_end: "<|audio_end|>",
+            time: "<|t_{:.2f}|>",
+            code_start: "<|code_start|>",
+            code_end: "<|code_end|>",
+            text_sep: "<|text_sep|>",
+        };
+        this.text_prompt = "{bos}\n{text_start}{words}{text_end}\n{audio_start}\n";
+        this.map_audio_tokens = this.get_audio_token_map();
+
+        this.languages = languages;
+    }
+
+    get_audio_token_map() {
+        const map = new Map();
+        for (let i = 0; i < 4100; ++i) {
+            const token = this.tokenizer.encode(this.special_tokens.audio_code.replace("{}", i), {
+                add_special_tokens: false,
+            })[0];
+            map.set(BigInt(token), i);
+        }
+        return map;
+    }
+
+    process_text(text, language) {
+        if (!this.languages.includes(language)) {
+            throw new Error(`Language ${language} not supported, supported languages are ${this.languages}`);
+        }
+        if (language !== "en") {
+            // TODO: Implement pre-processing for non-English languages
+            console.warn("Non-English languages are not fully supported yet, so the output may not be accurate.");
```

Comment:
This should probably throw an error for now, as it will fail to generate anything if the input contains non-Latin characters.